module/User/src/Model/User.php

```diff
@@ -152,13 +152,29 @@ public function getRoleNames()
     }
 
     /**
-     * Get the user's role ID.
-     *
      * @return string
      */
-    public function getRoleId()
+    public function getRoleId(): string
     {
-        return 'user_' . $this->getLidnr();
+        if (in_array('admin', $this->getRoleNames())) {
+            return 'admin';
+        }
+
+        // TODO: We could create an assertion for this. (member check for C4)
```

Comment:
I do not think we should make this an assertion. The assertion is an additional check for every `isAllowed()` that is performed in an execution, while the `Member` object is only created once during an execution. Hence making it an assertion would create overhead.